src/ui/source_editor_area.ui

```diff
@@ -16,18 +16,46 @@
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
+  <property name="minimumSize">
+   <size>
+    <width>0</width>
+    <height>220</height>
+   </size>
+  </property>
   <property name="windowTitle">
    <string>Form</string>
   </property>
-  <layout class="QVBoxLayout" name="verticalLayout">
+  <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1">
    <property name="spacing">
-    <number>0</number>
+    <number>3</number>
+   </property>
+   <property name="leftMargin">
+    <number>3</number>
+   </property>
+   <property name="topMargin">
+    <number>3</number>
    </property>
-   <property name="margin">
-    <number>0</number>
+   <property name="rightMargin">
+    <number>3</number>
+   </property>
+   <property name="bottomMargin">
+    <number>3</number>
    </property>
    <item>
+    <widget class="QLabel" name="label_editor">
+     <property name="text">
+      <string>Lua code:</string>
```

Comment:
Actually that place holder text actually caused me to raise a [bug on Qt](https://bugreports.qt.io/browse/QTBUG-51275) - anyway in some error messages it is necessary to describe where the error is, and in the specific case of Lua Condition triggers it is necessary to differentiate between a Lua error in a numbered "item" and one in the "Lua code:" area.

For the record the place holder only works for Qt 5.3 and later (I had to make the code to do it version dependent and I would, in hind sight, be happy to do away with it)... :man_shrugging:

Actually this might be an issue on the Buttons/Menu/Toolbar display, where there are two (large) and now adjustable (and if you get your wish to make things collapsible) widgets.  You suggested that we change the label for "CSS" so it now says "Style-sheet:", to then have another multi-line text entry box which would be unlabelled to my eye looks a bit disjoint from a common visual theme - though then the "Errors:" widget is also unlabelled. {OTOH that pops-up and hides on user request and is visually unrelated to the rest of the dialogue by having a black background compared to everything else being pale by default.}

It would also look "off" on the "Variables" editor where everything else has a name.

These are some of my thoughts on the aesthetics but I am willing to hear further input on this. :ear: